cjworkbench/i18n/views.py

```diff
@@ -15,14 +15,13 @@ def set_locale(request):
     
     Based on `django.views.i18n.set_language`
     """
-    next = request.META.get("HTTP_REFERER")
-    next = next and unquote(next)  # HTTP_REFERER may be encoded.
+    next = request.POST.get("next", "/")
     if not is_safe_url(
         url=next, allowed_hosts={request.get_host()}, require_https=request.is_secure()
     ):
         next = "/"
     if request.method == "POST":
```

Comment:
Nix this if-statement. It implies we've tested what happens when the request is _not_ `POST` ... and we haven't. (I don't really care what happens; and if it's really bad, I'll get an email about it.)